### PR TITLE
multiple frontend changes on multi-select lists

### PR DESCRIPTION
**Pull request by jsaalfeld**

### CODE REVIEW

**Comment by MaximilianKrambach** [comment]
A wonderful example for 'How to communicate'.

For the curious: implements a 'select all' button, and changes the multiselection from the whole data row to just the first column.

**Comment by jsaalfeld** [comment]
> changes the multiselection from the whole data row to just the first column.

not just the first column, but to the columns that have otherwise no secondary click functions.

File: client/src/app/site/users/components/user-list/user-list.component.ts [hunk 1]
```diff
@@ -203,11 +203,27 @@ export class UserListComponent extends ListViewBaseComponent<ViewUser> implement
...
     }
+
+    /**
+     * Sets the user present
+     *
+     * @param viewUser the viewUser Object
+     */
+    public async setPresent(viewUser: ViewUser): Promise<void> {
+        const user = viewUser.user;
+        user.is_present = !viewUser.user.is_present;
```
Comment (FinnStutzenstein) [inline client/src/app/site/users/components/user-list/user-list.component.ts]: either `viewUser.user.is_present = !viewUser.user.is_present;` and not `const ...` or `user.is_present = !user.is_present;`

File: client/src/app/site/mediafiles/components/mediafile-list/mediafile-list.component.ts [hunk 2]
```diff
@@ -270,6 +270,7 @@ export class MediafileListComponent extends ListViewBaseComponent<ViewMediafile>
      * @param file the select file to download
      */
     public singleSelectAction(file: ViewMediafile): void {
+        console.log('olo')
```
Comment (FinnStutzenstein) [inline client/src/app/site/mediafiles/components/mediafile-list/mediafile-list.component.ts]: please remove

File: client/src/app/site/assignments/assignment-list/assignment-list.component.html [hunk 3]
```diff
@@ -31,14 +32,23 @@
...
     </ng-container>
+    <!-- pahse column-->
     <ng-container matColumnDef="phase">
         <mat-header-cell *matHeaderCellDef mat-sort-header>Phase</mat-header-cell>
         <mat-cell *matCellDef="let assignment">
             <mat-chip-list>
                 <mat-chip color="primary" selected>{{ assignment.phase }}</mat-chip>
             </mat-chip-list>
-        </mat-cell>
+        </mat-cell><button mat-menu-item (click)="toggleMultiSelect()">
```
Comment (tsiegleauq) [inline client/src/app/site/assignments/assignment-list/assignment-list.component.html]: looks falsely aligned. Did that run through prettier? Usually prettier does not accept `</something><something>`

File: client/src/app/site/mediafiles/components/mediafile-list/mediafile-list.component.ts (hunk 2 above)
Comment (tsiegleauq) [inline client/src/app/site/mediafiles/components/mediafile-list/mediafile-list.component.ts]: missing "y" ;-)

File: client/src/app/site/users/components/user-list/user-list.component.html [hunk 4]
```diff
@@ -59,16 +60,14 @@
         <mat-header-cell *matHeaderCellDef mat-sort-header>Presence</mat-header-cell>
         <mat-cell *matCellDef="let user">
             <div *ngIf="user.is_active">
-                <mat-icon>check_box</mat-icon>
-                <span translate>Present</span>
+                <mat-checkbox (click)="setPresent(user)" [checked]="user.is_present"><span translate>Present</span></mat-checkbox>
```
Comment (tsiegleauq) [inline client/src/app/site/users/components/user-list/user-list.component.html]: thats too far to the right edge of the screen. Looks cut off

**Comment by jsaalfeld** [comment]
Worked through the annotations and added `deselectAll`

**Pull request merged**